pypeit/scripts/coadd_1dspec.py

```diff
@@ -135,43 +135,42 @@ def get_parser(cls, width=None):
-                                "   objid      -- is the object identifier. To determine the objids inspect the \n"
-                                "                 spec1d_*.txt files or run pypeit_show_1dspec spec1dfile --list\n"
-                                "\n")
+        parser.add_argument('coadd1d_file', type=str,
+                            help="R|File to guide coadding process. This file must have the "
+                                 "following format: \n\n"
+                                 "F|[coadd1d]\n"
+                                 "F|   coaddfile='output_filename.fits'\n"
+                                 "F|   sensfuncfile = 'sensfunc.fits' # Required only for Echelle\n"
+                                 "\n"
+                                 "F|   coadd1d read\n"
+                                 "F|     spec1dfile1 objid1\n"
+                                 "F|     spec1dfile2 objid2\n"
+                                 "F|     spec1dfile3 objid3\n"
+                                 "F|        ...    \n"
+                                 "F|   coadd1d end\n"
+                                 "\n OR the coadd1d read/end block can look like \n\n"
+                                 "F|  coadd1d read\n"
+                                 "F|     spec1dfile1 objid \n"
+                                 "F|     spec1dfile2 \n"
+                                 "F|     spec1dfile3 \n"
+                                 "F|     ...    \n"
+                                 "F|  coadd1d end\n"
+                                 "\n"
+                                 "That is the coadd1d block must either be a two column list of "
+                                 "spec1dfiles and objids, or you can specify a single objid for "
+                                 "all spec1dfiles on the first line\n\n"
+                                 "Where: \n"
+                                 "\n"
+                                 "spec1dfile: full path to a PypeIt spec1dfile\n\n"
+                                 "objid: the object identifier. To determine the objids inspect "
+                                 "the spec1d_*.txt files or run pypeit_show_1dspec spec1dfile "
+                                 "--list\n\n")
         parser.add_argument("--debug", default=False, action="store_true", help="show debug plots?")
-        parser.add_argument("--show", default=False, action="store_true", help="show QA during coadding process")
-        parser.add_argument("--par_outfile", default='coadd1d.par', help="Output to save the parameters")
+        parser.add_argument("--show", default=False, action="store_true",
+                            help="show QA during coadding process")
+        parser.add_argument("--par_outfile", default='coadd1d.par',
+                            help="Output to save the parameters")
         parser.add_argument("--test_spec_path", type=str, help="Path for testing")
```

Comment:
The --test_spec_path option here as well as the coadd1d_filelist method above on line 86 were added I think in a previous PR. I missed these going by. My opinion is that methods and options only used for testing should be in tests and not in the production code itself. It makes the argument lists longer and adds codes to the script that are not necessary for the scripts functioning. I agree these lines of code and options should exist, but in my opinion they should be in the tests not here.

Reply:
I imagine you noticed this, but, just to be clear, the only reason these lines changed is because I restricted the lines to be no longer than 100 characters.  So I would suggest this comment is beyond the scope of this PR.  Having said that, if there's something specific you want me to do (like remove some of these command-line options), I can do that.

Reply:
Hi @kbwestfall, I agree that this is beyond the scope of this PR and am not advocating any changes now. So you can go ahead and merge. However, I would like to discuss this at the telecon, because I personally really don't like to see a bunch of testing code in code that does not run tests. There are likely a diversity of opinions on this, so it is something we should discuss as a group.

Reply:
Sounds good.  I added an item on this to the 29 June agenda.  Merging this PR now.